sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs

```diff
@@ -2363,7 +2363,7 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
                     conditions: options?.OpenConditions,
                     accessTier: default,
                     progressHandler: default,
-                    operationName: default,
+                    operationName: "BlockBlobClient.OpenWriteInternal",
```

Comment:
This was causing problem in VS debugger
```
Unhandled Exception: System.ArgumentNullException: Value cannot be null.
Parameter name: key
at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
at Microsoft.IntelliTrace.TelemetryObserver.IntelliTraceObserver.IsEnabled(String telemetryName)
at System.Diagnostics.DiagnosticListener.IsEnabled(String name)
at Azure.Core.Pipeline.DiagnosticScope..ctor(String name, DiagnosticListener source) in C:\git\azure-sdk-for-net\sdk\core\Azure.Core\src\Shared\DiagnosticScope.cs:line 25
at Azure.Core.Pipeline.DiagnosticScopeFactory.CreateScope(String name) in C:\git\azure-sdk-for-net\sdk\core\Azure.Core\src\Shared\DiagnosticScopeFactory.cs:line 51
at Azure.Storage.Blobs.Specialized.BlockBlobClient.UploadInternal(Stream content, BlobHttpHeaders blobHttpHeaders, IDictionary`2 metadata, IDictionary`2 tags, BlobRequestConditions conditions, Nullable`1 accessTier, IProgress`1 progressHandler, String operationName, Boolean async, CancellationToken cancellationToken) in C:\git\azure-sdk-for-net\sdk\storage\Azure.Storage.Blobs\src\BlockBlobClient.cs:line 784
at Azure.Storage.Blobs.Specialized.BlockBlobClient.OpenWriteInternal(Boolean overwrite, BlockBlobOpenWriteOptions options, Boolean async, CancellationToken cancellationToken) in C:\git\azure-sdk-for-net\sdk\storage\Azure.Storage.Blobs\src\BlockBlobClient.cs:line 2358
at Azure.Storage.Blobs.Specialized.BlockBlobClient.OpenWrite(Boolean overwrite, BlockBlobOpenWriteOptions options, CancellationToken cancellationToken) in C:\git\azure-sdk-for-net\sdk\storage\Azure.Storage.Blobs\src\BlockBlobClient.cs:line 2273
at ConsoleApp1.Program.Main(String[] args) in C:\git\azure-sdk-for-net\sdk\storage\ConsoleApp1\Program.cs:line 52
at ConsoleApp1.Program.<Main>(String[] args)
```

Reply:
I think we should `string operationName = $"{nameof(BlockBlobClient)}.{nameof(OpenWrite)}";` on line 2344, use `operationName` to create the DiagnosticScope on the next line, and also refer to operationName here.